Makefile

```diff
@@ -130,4 +138,42 @@ ci_after_success:
 	cat ./coverage/lcov.info | ./node_modules/codecov/bin/codecov
 	./node_modules/nyc/bin/nyc.js report --reporter=lcov && ./node_modules/codecov/bin/codecov
 
-.PHONY: metadata.db ci ci_after_success
+# NOTE: this is executed on a RHEL 7 or CentOS 7 host which is
+# VM or bare metal. All required repositories must be configured, e.g.
+# https://copr.fedorainfracloud.org/coprs/g/weldr/lorax-composer/repo/epel-7/group_weldr-lorax-composer-epel-7.repo
+test_with_lorax_composer: rpm
+# install required RPM packages
+	sudo yum -y install lorax-composer cockpit-ws cockpit-kubernetes welder-web*.rpm
+
+# don't require Cockpit authentication when logging in
+	sudo /bin/sh -c 'echo -e "[Negotiate]\nCommand = /usr/libexec/cockpit-stub\n[WebService]\nShell = /shell/simple.html" > /etc/cockpit/cockpit.conf'
+	sudo setenforce 0
+
+# patch application to use UNIX socket with lorax-composer
+	sudo sed -i "s|welderApiPort=.*|welderApiPort='/run/weldr/api.socket';|" /usr/share/cockpit/welder/js/config.js
+	sudo systemctl restart cockpit
+
+# build e2e test images with increased timeouts
+	sed -i "s|waitforTimeout: 30000|waitforTimeout: 90000|" ./test/end-to-end/wdio.conf.js
+	sudo docker build -f Dockerfile.nodejs -t welder/web-nodejs:latest .
+	sudo docker build -f ./test/end-to-end/Dockerfile -t welder/web-e2e-tests:latest ./test/end-to-end/
+
+# execute lorax-composer in the background to serve as the API backend
+# making the socket accessible to the cockpit-ws group
+	sudo mkdir /recipes
+	sudo lorax-composer --group cockpit-ws /recipes &
+
+# wait for the backend to become ready
+	until sudo curl --unix-socket /run/weldr/api.socket http://localhost:4000/api/status | grep 'db_supported": true'; do \
```

Comment:
And here.

Reply:
This part here is executed on the RHEL host running lorax-composer and with curl-7.29.0-46.el7.x86_64 this option is not available. I will see of it is available for the above 2 places which are executed inside newer environments (Ubuntu/Travis).

Reply:
Turns out --connection-timeout isn't supported in none of our environments.

Reply:
How about

    for retry in `seq 60`; do curl [...] && exit 0; sleep 1; done; echo  "TIMED OUT" >&2; false

?

Reply:
Not against this but I think 60 seconds is not enough for lorax-composer. This will also depend on the local system that we happen to get for testing and the size/count of available repositories.

We can definitely fix this in a follow up commit though.